game-core/src/main/java/games/strategy/triplea/ui/TripleAFrame.java

```diff
@@ -614,7 +614,7 @@ public void leaveGame() {
       ((ClientGame) game).shutDown();
       // an ugly hack, we need a better
       // way to get the main frame
-      GameRunner.clientLeftGame();
+      new Thread(GameRunner::clientLeftGame).start();
```

Comment:
This is probably the only exception where things are made more asynchronous than before.
However this makes closing the window smooth, because previously we randomly waited for 100ms on the EDT so everything froze for this interval